### ISSUE TITLE
Http/Requests doesn't perform JSON validation on a JSON input

### ISSUE THREAD

**Issue by CaporalDead** [laravel/framework#31943]
- Laravel Version: >=5.4.*
- PHP Version: >=5.6 (depending on laravel version)

### Description:

It seems that Laravel doesn't check if a supposed JSON input is valid or not. Is this a wanted behavior or a bug affecting all Laravel versions ?

Thanks for your feedback :)

### Steps To Reproduce:

On any Laravel version since 5.4 (as far as I know).

Initialize a new laravel project using the composer create-project command
Initialize a route like so on any method (GET/POST/PUT) :

```php
<?php

Route::post('/api/test', function () {
    return request()->input();
});
```

Using your preferred tool, call your API : 

#### Valid JSON input

POST http://127.0.0.1:8000/api/test
```json
{
    "foo": "bar"
}
```

##### Expected and got response : 

`200 Ok`
```json
{
    "foo": "bar"
}
```

#### Invalid JSON input

POST http://127.0.0.1:8000/api/test
```json
{
    "foo": "bar",
}
```

##### Expected response (something similar to) : 

`4xx`
```json
{
    "message": "Bad JSON input"
}
```

##### Actual response :

`200 Ok`
```json
[]
```

The first problem here is that the `Http\Request::json` method try to decode the given JSON but doesn't check the result of `json_last_error`. 

A simple patch could be to add some error checking as followed (or something similar) : 

```php
    public function json($key = null, $default = null)
    {
        if (! isset($this->json)) {
            $this->json = new ParameterBag((array) json_decode($this->getContent(), true));
        }

        if (JSON_ERROR_NONE !== json_last_error()) {
            throw new RuntimeException('Bad input JSON format');
        }

        if (is_null($key)) {
            return $this->json;
        }

        return data_get($this->json->all(), $key, $default);
    }
```

**Comment by driesvints** [comment]
Hey there,

Unfortunately we don't support this version anymore. Please check out [our support policy](https://laravel.com/docs/releases) on which versions we are currently supporting. Can you please try to upgrade to the latest version and see if your problem persists? We'll help you out and re-open this issue if so.

Thanks!

**Comment by CaporalDead** [comment]
@driesvints as I said it's affectinf all versions from 5.4 at least including last version from fresh install.

Laravel Version: **>=**5.4.*

**Comment by driesvints** [comment]
@CaporalDead this seems like an issue with your app so please try a support channel.

**Comment by JMartelot** [comment]
Hi @driesvints, thanks for your reply.

I have done some tests too, with different laravel versions and I reproduce the usecase, indeed when your payload is malformed, Laravel don't take care to validate it.

**Comment by CaporalDead** [comment]
@driesvints I don't understand your answer, there's nothing specific to my app. You can reproduce this behavior on any fresh install of Laravel following the informations I provided in my first post.

**Comment by matheusb-comp** [comment]
This is related to #27754, #10257, and #5022.
Also pull request #15228 has already been rejected.

**Comment by Kwaadpepper** [comment]
If anyone is viewing this, I think this is a flaw. I am starting to think that Laravel request are really "generic". Meaning they will just work no matter what user input is (untrusted as always). But while using request validator, this is really neat. It will permit forging requests in a weird way. There are fewer ways for us to enforce controls.

Is mentionned, I am using this solution as a middleware on my api to make sure only valid json request are send.

      <?php
      
      namespace App\Http\Middleware;
      
      use Illuminate\Http\Request;
      
      class JsonOnly
      {
          /**
           * Handle an incoming request.
           *
           * @param  \Illuminate\Http\Request $request
           * @param  \Closure                 $next
           * @return mixed
           */
          public function handle(Request $request, \Closure $next)
          {
              if (
                  !Helper::isRequestForApiV1($request) and
                  (!$request->isJson() or
                      !$request->expectsJson() or
                      is_null(json_decode($request->getContent())))
              ) {
                  abort(400, 'Invalid JSON');
              }
      
              return $next($request);
          }
      }

This will make sure requests have :
- ``Accept: application/json``
- ``Content-Type: application/json``
- Body : a json encoded string and when empty it must have ``"[]"``
